server/src/main/java/io/littlehorse/server/streams/util/POSTStreamObserver.java

```diff
@@ -28,11 +36,20 @@ public POSTStreamObserver(
         this.shouldComplete = shouldComplete;
         this.internalComms = internalComms;
         this.command = command;
+        this.commandStartedAt = new Date();
+        this.timeoutAt = new Date(commandStartedAt.getTime() + successResponseTimeout.toMillis());
     }
 
     @Override
     public void onError(Throwable t) {
-        ctx.onError(t);
+        final boolean isRetryable = t instanceof StatusRuntimeException grpcRuntimeException
+                && grpcRuntimeException.getStatus().getCode().equals(Status.UNAVAILABLE.getCode());
+        final boolean retryOneMoreTime = timeoutAt.compareTo(new Date()) > 0;
+        if (isRetryable && retryOneMoreTime) {
+            internalComms.waitForCommand(command, this);
```

Comment:
Hmm. This seems like a busy retry. Fine for now I think but in the future we'll have to implement a backoff mechanism.

Reply:
https://github.com/littlehorse-enterprises/littlehorse/issues/873